core/src/server/rpc_module.rs

```diff
@@ -786,9 +746,40 @@ impl SubscriptionSink {
 		self.inner_send(msg).map_err(Into::into)
 	}
 
+	/// Consume the sink by passing a stream to be sent via the sink.
+	pub async fn add_stream<S, T>(mut self, mut stream: S)
+	where
+		S: Stream<Item = T> + Unpin,
+		T: Serialize,
+	{
+		let mut close_stream = self.close.clone();
```

Comment:
this is hack; to workaround borrowing issues when we call `self.send` below.

we may refactor this to split this up to several types or something....

Reply:
I think we need plenty of comments in this method, it requires some pretty deep understanding of the architecture and I think we should capture it here.